### DISCUSSION TITLE
Heroku Buildpack JAVA JDK cache

### DISCUSSION THREAD

**Discussion by Milanoss** [dokku/dokku#4565] [Q&A]
It seems that JDK is not cached during deploy JAVA Spring boot app. I can see "curl ...s3....jdk" in processes during deploy and it takes a lot of time to deploy small java app. Do you have the same experience? Or do I need some configuration to do caching of JDK? I can see some caching in Heroku JAVA Buildpack sources but seems it does not work :(

**Comment by josegonzalez** [comment]
If you are using a buildpack to deploy, its the responsibility of the buildpack to perform caching. In our case, we use the Heroku buildpacks, so whatever they do is what we respect. My guess is that the buildpacks reinstall those dependencies. Without the entire log output from the deploy, it would be hard to guess at what is going on.
